cmd/kubeadm/app/util/version_test.go

```diff
@@ -121,9 +121,10 @@ func TestVersionFromNetwork(t *testing.T) {
 		"stable-1.3": {"v1.3.10", "v1.3.10", false, false},
 		"latest":     {"v1.6.0-alpha.0", "v1.6.0-alpha.0", false, false},
 		"latest-1.3": {"v1.3.11-beta.0", "v1.3.11-beta.0", false, false},
-		"empty":      {"", currentVersion, true, false},
-		"garbage":    {"<?xml version='1.0'?><Error><Code>NoSuchKey</Code><Message>The specified key does not exist.</Message></Error>", currentVersion, true, false},
-		"unknown":    {"The requested URL was not found on this server.", currentVersion, true, false},
+		"empty":      {"", "", true, true},
+		"garbage":    {"", "", true, true},
+		"unknown":    {"", "", true, true},
```

Comment:
instead of these 3 that are essentially the same test case, IMO we should have a list of bad inputs to the regexp to keep it from breaking when changed in the future.

e.g. see my go playground tests:
https://play.golang.org/p/zprnyzsKDnV